pliers/extractors/audio.py

```diff
@@ -488,4 +495,126 @@ class PercussiveExtractor(LibrosaFeatureExtractor):
+        are to be returned. If None, all labels (or top_n) are returned.
+        The full list of labels is available in the audioset/yamnet 
+        repository (see yamnet_class_map.csv).
+    weights_path (optional): full path to model weights file. If not provided,
+        weights from pretrained YAMNet module are used.
+    yamnet_kwargs (optional): Optional named arguments that modify input 
+        parameters for the model (see params.py file in yamnet repository)
+    '''
+
+    _log_attributes = ('hop_size', 'top_n', 'labels', 'weights_path',
+                       'yamnet_kwargs')
+
+    def __init__(self, hop_size=0.1, top_n=None, labels=None,
+                 weights_path=None, yamnet_path=None, **yamnet_kwargs):
+        if yamnet_path is None:
+            yamnet_path = YAMNET_PATH
+        try:
+            sys.path.insert(0, str(yamnet_path))
+            self.yamnet = attempt_to_import('yamnet')
+            verify_dependencies(['yamnet'])
+        except MissingDependencyError: 
+            msg = ('Yamnet could not be imported. To download and set up '
+                  'yamnet, run:\n\tpython -m pliers.support.setup_yamnet')
+            raise MissingDependencyError(dependencies=None,
+                                         custom_message=msg)
+        verify_dependencies(['tensorflow'])
+        if top_n and labels:
+            raise ValueError('Top_n and labels are mutually exclusive '
+                             'arguments. Reinstantiate the extractor setting '
+                             'top_n or labels to None (or leaving it '
+                             'unspecified).')
+
+        MODULE_PATH = path.dirname(self.yamnet.__file__)
+        LABELS_PATH = path.join(MODULE_PATH, 'yamnet_class_map.csv')
+        self.weights_path = weights_path or path.join(MODULE_PATH, 'yamnet.h5')
+        self.hop_size = hop_size
+        self.yamnet_kwargs = yamnet_kwargs or {}
+        self.params = self.yamnet.params
+        self.params.PATCH_HOP_SECONDS = hop_size
+        for par, v in self.yamnet_kwargs.items():
+            setattr(self.params, par, v)
+        if self.params.PATCH_WINDOW_SECONDS != 0.96:
+            logging.warning('Custom values for PATCH_WINDOW_SECONDS were '
+                'passed. YAMNet was trained on windows of 0.96s. Different '
+                'values might yield unreliable results.')
+
+        self.top_n = top_n
+        all_labels = pd.read_csv(LABELS_PATH)['display_name'].tolist()
+        if labels is not None:
+            missing = list(set(labels) - set(all_labels))
+            labels = list(set(labels) & set(all_labels))
+            if missing:
+                logging.warning(f'Labels {missing} do not exist. Dropping.')
+            self.labels = labels
+            self.label_idx = [i for i, l in enumerate(all_labels) 
+                              if l in labels]
+        else:
+            self.labels = all_labels
+            self.label_idx = range(len(all_labels))
```

Comment:
`range()` will return a generator, so depending on what you're doing with this later (or just to play it safe), you might want to convert it to a list at this point. If it's just passed through to `ExtractorResult` then it's probably fine.

Reply:
converted to list here.